### PR TITLE
feat: adds in capacity for using images, token names and symbols matc…

### PR DESCRIPTION
**Pull request by R-K-H**
…hing our patterns for metadata

### CODE REVIEW

File: packages/indexer/src/indexers/autocrat/autocrat-proposal-indexer.ts [hunk 1]
```diff
@@ -117,13 +135,33 @@ export const AutocratProposalIndexer: IntervalFetchIndexer = {
...
           const metadata = await enrichTokenMetadata(token, provider);
           const storedMint = await getMint(provider.connection, token);
+          // NOTE: THIS IS ONLY FOR PROPOSALS AND ONLY FOR BASE / QUOTE CONDITIONAL
+          // TODO: Need to update this for the symbol, name, and image_url base on pass / fail
+          const isQuote = [quoteFail, quotePass].includes(token);
+          const isFail = [quoteFail, baseFail].includes(token);
+          // TODO: Need to get metadata for the base token
+          // TODO: This needs to be refactored
+          let noMetadataName = isFail ? `Proposal ${proposal.account.number}: f${baseTokenMetadata.symbol}` : `Proposal ${proposal.account.number}: p${baseTokenMetadata.symbol}`;
+          let noMetadataSymbol = isFail ? `f${baseTokenMetadata.symbol}` : `p${baseTokenMetadata.symbol}`;
```
Comment (metaproph3t) [inline packages/indexer/src/indexers/autocrat/autocrat-proposal-indexer.ts:145]: recommend rename to `defaultMetadataName` and `defaultMetadaSymbol`

File: packages/indexer/src/indexers/autocrat/autocrat-proposal-indexer.ts [hunk 2]
```diff
@@ -117,13 +135,33 @@ export const AutocratProposalIndexer: IntervalFetchIndexer = {
...
+          let imageUrl;
+          if(isQuote){
+            noMetadataName = isFail ? `Proposal ${proposal.account.number}: fUSDC` : `Proposal ${proposal.account.number}: pUSDC`;
+            noMetadataSymbol = isFail ? `fUSDC` : `pUSDC`;
+          }
```
Comment (metaproph3t) [inline packages/indexer/src/indexers/autocrat/autocrat-proposal-indexer.ts:150]: this is kinda weird branching logic, I'd do
`let imageUrl, defaultMetadataName, defaultMetadataSymbol;`
and then follow that with an if else on `isQuote`

Comment (metaproph3t) [reply packages/indexer/src/indexers/autocrat/autocrat-proposal-indexer.ts:150]: or, if you want to be clever:
```
let passOrFailPrefix = isFail ? "f" : "q";
let baseSymbol = isQuote ? "USDC" : baseTokenMetadata.symbol;
let defaultSymbol = passOrFailPrefix + baseSymbol;
let defaultName = `Proposal ${proposal.account.number}: ${defaultSymbol}`

File: packages/indexer/src/indexers/autocrat/autocrat-proposal-indexer.ts [hunk 3]
```diff
@@ -117,13 +135,33 @@ export const AutocratProposalIndexer: IntervalFetchIndexer = {
...
 
+          if(dao && daoDetails) {
+            imageUrl = isFail ? daoDetails[0].fail_token_image_url : daoDetails[0].pass_token_image_url;
+            if(isQuote){
+              imageUrl = isFail ? "https://imagedelivery.net/HYEnlujCFMCgj6yA728xIw/6b1ce817-861f-4980-40ca-b55f28f21400/public" : "https://imagedelivery.net/HYEnlujCFMCgj6yA728xIw/f236a0ca-5d7c-4f4a-ca8a-52eb9d72ef00/public";
+            }
```
Comment (metaproph3t) [inline packages/indexer/src/indexers/autocrat/autocrat-proposal-indexer.ts:156]: again with the weird branching logic

**Review by metaproph3t — APPROVED** [review]
some suggestions, but LGTM

File: packages/indexer/src/indexers/autocrat/autocrat-proposal-indexer.ts (hunk 3 above)
Comment (R-K-H) [reply packages/indexer/src/indexers/autocrat/autocrat-proposal-indexer.ts:156]: I'm not sure what you mean by this, happy to adjust, just not clear on another way to think about it. There's four strings dependent on if it's a USDC or TOKEN and if it's pass or fail. The above example you had made sense as it was simple string appending / combination.

**Pull request merged**